lib/provisioning/Provisioner.js

```diff
@@ -91,62 +159,195 @@ Provisioner.prototype.isProvisionRequest = function(req) {
+        }
+        catch (err) {
+            log.error(`Error retrieving power levels (${err.data.error})`);
+        }
+
+        if (!powerState) {
+            throw new Error('Could not retrieve your power levels for the room');
+        }
+
+        let actualPower = 0;
+        if (powerState.users[userId] !== undefined) {
+            actualPower = powerState.users[userId];
+        }
+        else if (powerState.users_default !== undefined) {
+            actualPower = powerState.users_default;
+        }
+
+        let requiredPower = 50;
+        if (powerState.events["m.room.power_levels"] !== undefined) {
+            requiredPower = powerState.events["m.room.power_levels"]
+        }
+        else if (powerState.state_default !== undefined) {
+            requiredPower = powerState.state_default;
+        }
+
+        if (actualPower < requiredPower) {
+            throw new Error('User does not possess high enough power level');
+        }
+
+        // (IRC) Check that op's nick is actually op
+        log.info(`Check that op's nick is actually op`);
+        let botClient = yield this._ircBridge.getBotClient(server);
+        yield botClient.joinChannel(ircChannel, key);
+        log.info(`Bot joined channel ${ircChannel}`);
+        let info = yield botClient.getOperators(ircChannel);
+
+        if (info.nicks.indexOf(opNick) == -1) {
+            let knownOps = info.operatorNicks.join(', ');
+            throw new Error(`Provided user is not in channel ${ircChannel}. ` +
+                            `Known ops in this channel: ${knownOps}`);
+        }
+
+        if (info.operatorNicks.indexOf(opNick) == -1) {
+            let knownOps = info.operatorNicks.join(', ');
+            throw new Error(`Provided user is not an op of ${ircChannel}. ` +
+                            `Known ops in this channel: ${knownOps}`);
+        }
+
+        // (Matrix) update room state
+        log.info(`Sending m.room.bridging`);
+        // State key for m.room.bridging
+        let skey = `irc://${ircDomain}/${ircChannel}`;
+
+        // Send pending m.room.bridging
+        yield this._updateBridgingState(roomId, userId, 'pending', skey);
 
-Provisioner.prototype._validate = function(actual, parameterName) {
-    let valid = parameterValidation[parameterName];
+        // (IRC) Ask operator for authorisation
+        // Time that operator has to respond before giving up
+        let timeoutSeconds = this._requestTimeoutSeconds;
 
-    if (!valid) {
-        throw new Error(
-            `Parameter name not recognised (${parameterName}).`
-        );
+        // Deliberately not yielding on this so that 200 OK is returned
+        log.info(`Contacting operator`);
+        this._contactOperator(
+            botClient, server, opNick, ircChannel,
+            roomId, userId, skey, timeoutSeconds);
     }
+);
+
+Provisioner.prototype._contactOperator = Promise.coroutine(
```

Comment:
Misleading function name. It doesn't *just* contact the operator, it also then updates room state and more importantly, **create the link** if the contact was succesful.